packages/core/src/InputObserverController.ts

```diff
@@ -19,63 +21,100 @@ type MemoryLinksCountObserver = {
 const ThrottleMS: number = 100;
 
 export class InputObserverController {
-
-  public executionObservers: ExecutionObserver[] = [];
-
   private items$ = new Subject<MemoryItemObserver>();
   private links$ = new Subject<MemoryLinksCountObserver>();
   private observerMap: Map<string, Subscription> = new Map();
 
+  // TODO: In the future, consider using indexDB or JSON Lines instead of variables for data storage
```

Comment:
Perhaps a more powerful Map alternative can be achieved using dependency injection in the future?